File: client/src/game/conf/MailConfig.ts
class MailConfig {
    private static _dic:Dictionary<MailConfig> = new Dictionary<MailConfig>();
    private static _arr:Array<MailConfig> = new Array<MailConfig>();

	 public Id :number ;
     public MailId :number ;
    
     public MailType :number ;
     public SubjectText :number ;
    
     public ContentText :number ;
    
     public SenderText :number ;
    
     public Icon :number ;
     public RewardItemIds : Array<number>  =new Array<number>();
     public RewardNum : Array<number>  =new Array<number>();
     public SaveTime :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<MailConfig>{
        return  MailConfig._dic;
    }
    public static get Arr():Array<MailConfig>{
        return MailConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:MailConfig = <MailConfig>json[i];
				MailConfig._arr.push(config);
				MailConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("MailConfig loader fail!");
        }
    }
}

File: tools/conf/output/go/MailConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type MailConfigElem struct {
	Id int
	MailId int
	MailContent string
	MailType int
	SubjectText int
	SubjectDes string
	ContentText int
	ContentDes string
	SenderText int
	SenderDes string
	Icon int
	RewardItemIds []int
	RewardNum []int
	SaveTime int
	
}

type MailConfig0 struct {
	Dic map[int]*MailConfigElem
	Arr []*MailConfigElem
}

var (
	MailConfig MailConfig0
)

func loadMailConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	MailConfig.Dic = make(map[int]*MailConfigElem)
	MailConfig.Arr = make([]*MailConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &MailConfig.Arr)
	for _,conf := range MailConfig.Arr{
		MailConfig.Dic[conf.Id] = conf
	}
	
}
